### ISSUE TITLE
"Remove and Sort Usings on save" feature not working on some projects

### ISSUE THREAD

**Issue by justcla** [Microsoft/VS-PPT#40]
Travis Illig reported:
"In a project using the new DNX build system targeting dnx46, the "Remove and Sort Usings on save" PowerCommand doesn't actually do anything on document save for any of the C# class files in the project. Same thing for "Format document on save" - saving does not result in the document being formatted.

The features do work as expected for other project types."

**Comment by justcla** [comment]
Tipfom reported:
"I'm having a similiar problem with Xamarin Shared Assets Librarys."

**Comment by justcla** [comment]
Blandau006 reported:
"I'm experiencing the same issue DNX projects."

**Comment by justcla** [comment]
LarryDavidJr reported:
"I don't even get the option on the right click, is that the same as everyone else?  That could just be an issue with 'project types' needing to be added to the source, but the source for this function does not appear to be in the public domain yet.  so how about it MS, either open up the source for that part, or fix it for .NET Core RC2!"

**Comment by justcla** [comment]
Zsolt Miskolczi reported:
"I have the same problem. Both of the Remove and sort unused using on Save and the Format document on save commands does not work on .net core projects."

**Comment by rezamohamed** [comment]
I can't seem to get the simple remove unused savings on Save or anything working with ASP.NET Core.

**Comment by VaclavElias** [comment]
Same problem. It doesn't work on any ASP.NET Core project I have.

**Comment by HADB** [comment]
Same as @VaclavElias.  It doesn't work on my ASP.NET Core project.

**Comment by VaclavElias** [comment]
Community, was anyone able to locate in this source code where "Format document on save" and "Remove and Sort Usings on save" is?

**Comment by whoiskevinrich** [comment]
I am having the same problem. I have these observations to report:

Given a solution with both ASP.NET Core and ASP.NET 4.6.1 projects:
- The feature works when working with the ASP.NET 4.6.1 project
- The feature does not work when working with the ASP.NET Core project.
- Both projects are class libraries targeting the 4.6.1 framework

**Comment by PeterKottas** [comment]
Same here everybody. Took a while to find this issue but at least I know it's only me.

**Comment by coderabsolute** [comment]
For me, it doesn't work on "Core Projects" specifically.

**Comment by THammond9** [comment]
Several of my co-workers and I am having this problem in ASP.NET projects. This option does seem to work for us if the warning level on the project is changed to 4. In fact, if any project does not have a warning level of 4 "remove and sort" does not work at all.) [see this stackoverflow question](http://stackoverflow.com/questions/32698011/remove-unnecessary-usings-not-working-in-visual-studio-2015). Also I don't know if this is a problem with PPT or VS2015. The question above does not talk about PPT and I used the built in VS command to remove unused usings.

**Comment by VaclavElias** [comment]
It doesn't work for me also on Visual Studio 2017!

**Comment by justcla** [comment]
@THammond9 "Remove and Sort Usings" (Lightbulb feature) as described in the stack overflow question is a base VS feature. PPT provides a new feature, which is an option to enable that will "Remove and Sort Usings on Save". When activated, it will apply the Remove and Sort Usings automatically when the file is saved.

**Comment by justcla** [comment]
@VaclavElias Are you using .Net Core projects? Do you have ReSharper installed?

**Comment by VaclavElias** [comment]
Yes, I tried it on .Net Core projects and also Asp.Net MVC 5 or Console Apps  and I do have ReSharper installed. The Lightbulb feature is working but PPT "Remove and Sort Usings on Save" is not working.

**Comment by THammond9** [comment]
@justcla The solution mentioned in the stackoverflow question fixes PPT "Remove and Sort Usings on Save" for me. The issue with the warning level applies to more than just ASP.NET projects. I encourage you to try it yourself:

- Install PPT 2015
- Turn on "Remove and Sort Usings on Save" in PPT
- Create a new C# project, ASP.NET, Library, console, or winform app
- Add redundant usings and save and see that PPT works
- Change the warning level to less and 4
- Add redundent usings and save and see that PPT does not work

**Comment by justcla** [comment]
Thanks!
Does that mean that with the right warning levels set, it will work for .NET Core projects?

**Comment by VaclavElias** [comment]
I will test it with PPT 2017 on my .NET Core projects tomorrow :)

**Comment by justcla** [comment]
The new version of [Power Commands](https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.PowerCommandsforVisualStudio) (v15.0.4) uploaded last night contains an enhancement to enable Remove and Sort Usings on Save to work with .Net Core projects - and also to work with VB in both regular .NET and .NET Core.
Thanks to @olegtk.

**Comment by VaclavElias** [comment]
Thanks @justcla for letting us know.

Ok, I cannot confirm if it was fixed or not but with ReSharper it still doesn't work unless

a) ReSharper is Suspended
b) Used this work around https://github.com/Microsoft/VS-PPT/issues/81#issuecomment-293831138

For now I am happy with b) so I get a few seconds saved for each file :)

Thanks!
